controllers/kubeadmconfig_controller_test.go

```diff
@@ -661,3 +868,18 @@ func newControlPlaneInitKubeadmConfig(machine *capiv1alpha2.Machine, name string
 func stringPtr(s string) *string {
 	return &s
 }
+
+func newFakeSecretFactory() SecretsClientFactory {
```

Comment:
This should return a concrete type.

The compiler will yell if FakeSecretFactory does not satisfy SecretsClientFactory.